src/pyhf/__init__.py

```diff
@@ -44,44 +44,69 @@ def set_backend(backend, custom_optimizer=None):
+        >>> pyhf.set_backend(b"pytorch", precision="64b")
         >>> pyhf.tensorlib.name
         'pytorch'
+        >>> pyhf.tensorlib.precision
+        '64b'
         >>> pyhf.set_backend(pyhf.tensor.numpy_backend())
         >>> pyhf.tensorlib.name
         'numpy'
+        >>> pyhf.tensorlib.precision
+        '64b'
 
     Args:
         backend (`str` or `pyhf.tensor` backend): One of the supported pyhf backends: NumPy, TensorFlow, PyTorch, and JAX
         custom_optimizer (`pyhf.optimize` optimizer): Optional custom optimizer defined by the user
+        precision (`str`): Floating point precision to use in the backend: ``64b`` or ``32b``. Default is backend dependent.
 
     Returns:
         None
     """
     global tensorlib
     global optimizer
 
+    _valid_precisions = ["32b", "64b"]
+    backend_kwargs = {}
+
+    if isinstance(precision, (str, bytes)):
+        if isinstance(precision, bytes):
+            precision = precision.decode("utf-8")
+        precision = precision.lower()
+
     if isinstance(backend, (str, bytes)):
         if isinstance(backend, bytes):
             backend = backend.decode("utf-8")
         backend = backend.lower()
+
+        if precision is not None:
+            backend_kwargs["precision"] = precision
+
         try:
-            backend = getattr(tensor, "{0:s}_backend".format(backend))()
+            backend = getattr(tensor, f"{backend:s}_backend")(**backend_kwargs)
         except TypeError:
             raise InvalidBackend(
-                "The backend provided is not supported: {0:s}. Select from one of the supported backends: numpy, tensorflow, pytorch".format(
-                    backend
-                )
+                f"The backend provided is not supported: {backend:s}. Select from one of the supported backends: numpy, tensorflow, pytorch"
             )
 
-    _name_supported = getattr(tensor, "{0:s}_backend".format(backend.name))
+    _name_supported = getattr(tensor, f"{backend.name:s}_backend")
     if _name_supported:
         if not isinstance(backend, _name_supported):
             raise AttributeError(
-                "'{0:s}' is not a valid name attribute for backend type {1}\n                 Custom backends must have names unique from supported backends".format(
-                    backend.name, type(backend)
-                )
+                f"'{backend.name:s}' is not a valid name attribute for backend type {type(backend)}\n                 Custom backends must have names unique from supported backends"
             )
+    if backend.precision not in _valid_precisions:
+        raise Unsupported(
+            f"The backend precision provided is not supported: {backend.precision:s}. Select from one of the supported precisions: {', '.join([str(v) for v in _valid_precisions])}"
+        )
+    # If "precision" arg passed, it should always win
+    # If no "precision" arg, defer to tensor backend object API if set there
+    if precision is not None:
```

Comment:
I don't understand when we get to this point. It seems like duplciated code.

Reply:
@kratsg The example I gave in the PR body hits this:

```python
>>> import pyhf
>>> pyhf.set_backend(pyhf.tensor.pytorch_backend(precision="32b"), precision="64b")
```